cmake/autocoder/deployment_validation.cmake

```diff
@@ -0,0 +1,42 @@
+####
+# autocoder/deployment_validation.cmake
+#
+# An autocoder that performs validation steps on the deployment to ensure that it is well-formed. This will produce
```

Comment:
It could be good to add a note to clarify that this is not actually going to generate any files, and that it's simply using the autocoder logic in order to be fed all the .fpp files. I can see a new developer look at this and wonder what's happening and why this is an autocoder.

Reply:
That is wise.  I will fix this.